qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,223 @@
+    ) -> None:
+        """Helper function implementing the kernelized Pegasos algorithm to fit the SVM for both the
+        pre computed and non pre computed version.
+        Args:
+            X: shape (x_samples, s), train features
+            y: shape (x_samples) train labels
+            precomputed_kernel: shape (x_samples, x_samples) optional pre computed kernel matrix
+
+        Raises:
+            ValueError:
+                - X and/or y have the wrong shape
+                - X and y have incompatible dimensions
+                - Pre-computed kernel matrix precomputed_kernel has the wrong shape and/or dimension
+                - y contains incompatible labels
+        """
+        # check whether the data has the right format
+        if np.ndim(X) != 2:
+            raise ValueError("X has to be a 2D array")
+        if np.ndim(y) != 1:
+            raise ValueError("y has to be a 1D array")
+        if X.shape[0] != y.shape[0]:
+            raise ValueError("X and y have to contain the same number of samples")
+        if len(np.unique(y)) != 2:
+            raise ValueError("Only binary classification is supported")
+        if (precomputed_kernel is not None) and (
+            not precomputed_kernel.shape == (X.shape[0], X.shape[0])
+        ):
+            raise ValueError(
+                f"precomputed_kernel has the wrong shape {precomputed_kernel.shape}, \
+                it should be {(X.shape[0], X.shape[0])}"
+            )
+
+        # the algorithm works with labels in {+1, -1}
+        self._label_pos = np.unique(y)[0]
+        self._label_neg = np.unique(y)[1]
+        self._label_dict[self._label_pos] = +1
+        self._label_dict[self._label_neg] = -1
+
+        # the training data is later needed for prediction
+        self._x_train = X
+        self._y_train = y
+
+        # empty dictionaries to represent sparse arrays
+        self._alphas = {}
+        kernel: Dict[Tuple, np.ndarray] = {}
+
+        t_0 = datetime.now()
+        for step in range(1, self._num_steps + 1):
+            i = np.random.randint(0, len(y))
+
+            value = 0.0
+            for j in self._alphas:  # only loop over the non zero alphas
+                if precomputed_kernel is None:
+                    kernel[(i, j)] = kernel.get((i, j), self._quantum_kernel.evaluate(X[i], X[j]))
+                    value += (
+                        self._alphas[j]
+                        * self._label_dict[y[j]]
+                        * (kernel[(i, j)] + self._kernel_offset)
+                    )
+                else:
+                    value += (
+                        self._alphas[j]
+                        * self._label_dict[y[j]]
+                        * (precomputed_kernel[i, j] + self._kernel_offset)
+                    )
+
+            if (self._label_dict[y[i]] * self.C / step) * value < 1:
+                self._alphas[i] = self._alphas.get(i, 0) + 1
```

Comment:
If you can add a few comments on the implementation here would be nice.

Reply:
See 0a3726d0bc75a6b4512e542066bc433cb0eefa64